fahrschein/src/main/java/org/zalando/fahrschein/ManagedCursorManager.java

```diff
@@ -73,8 +72,8 @@ void setStreamId(String streamId) {
     private final ObjectMapper objectMapper;
     private final Map<String, SubscriptionStream> streams;
 
-    public ManagedCursorManager(URI baseUri, RequestFactory clientHttpRequestFactory, AccessTokenProvider accessTokenProvider) {
-        this(baseUri, wrapClientHttpRequestFactory(clientHttpRequestFactory, accessTokenProvider), true);
+    public ManagedCursorManager(URI baseUri, RequestFactory clientHttpRequestFactory, AuthorizationProvider authorizationProvider) {
```

Comment:
If you want to be paranoid about binary compatibility there should be a separate overload. Could lead to problems when you have a transitive dependency compiled against the older version, however unlikely that is.

Reply:
I'd take the risk her, fahrschein is on `0.x.x` either way in the end :wink:

Reply:
> transitive dependency compiled against the older version

Fahrschein is not really used in other libraries, is it?

Reply:
Github is telling me it is used by [nakadi-producer-spring-boot-starter](https://github.com/zalando-nakadi/nakadi-producer-spring-boot-starter) and [fahrschein-spring-boot-starter](https://github.com/jbellmann/fahrschein-spring-boot-starter). There might be others that are non-public. Anyway, it should fail rather loudly during testing if there is an issue.

Reply:
These two 👆 are mostly authored by @ePaul